src/rust/ensogl/lib/core/src/data/color/data.rs

```diff
@@ -291,3 +291,11 @@ impl<C:Default> Default for Alpha<C> {
         Self {alpha,opaque}
     }
 }
+
+impl<C> Alpha<C> {
+    /// Return the color with a multiplied alpha channel.
+    pub fn multiply_alpha(self, alpha:f32) -> Color<Self> {
+        let alpha = self.alpha * alpha;
+        Color(Alpha{alpha, opaque: self.opaque })
+    }
```

Comment:
lets add here a function mod_alpha that takes a lambda. If you want you can of course keep this one as a shortcut over the mod_alpha one (that one is in-place and `with_mod_alpha` is not in place)